crates/iceberg-catalog/src/service/storage/hdfs.rs

```diff
@@ -0,0 +1,91 @@
+// use std::{
+//     collections::HashMap,
+//     str::FromStr,
+//     sync::{Arc, OnceLock},
+// };
+
+use iceberg_ext::configs::{table::TableProperties, Location, ParseFromStr};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    api::iceberg::v1::DataAccess,
+    service::storage::{
+        error::TableConfigError, StoragePermissions, StorageType, TableConfig, ValidationError,
+    },
+};
+
+#[derive(Debug, Eq, Clone, PartialEq, Serialize, Deserialize, utoipa::ToSchema)]
+#[serde(rename_all = "kebab-case")]
+pub struct HdfsProfile {
+    pub url: String,
+    pub prefix: String,
+}
```

Comment:
We should add a `Hashmap<String,String>` and pass it into the hdfs native client so that these [options](https://github.com/Kimahriman/hdfs-native?tab=readme-ov-file#supported-hdfs-settings) get exposed